cmake-naming.md

```diff
@@ -0,0 +1,36 @@
+## CMake targets for Mbed "targets"
+
+### targets.json naming scheme
+
+- `MCU_STM32L4` - base target
+- `DISCO_L4R9I` - final target - a user board or could be custom target, inherits from MCU_  target
+
+The targets.json properties use capital letters with underscore but not the our config. Probably to distinguish these from other regular directories (the initial Mbed SDK had it this way).
+
+### CMake naming scheme
+
+All Mbed OS CMake targets are prefixed with `mbed-` to avoid clashes with user's targets (out of the tree targets): `mbed-ble`, `mbed-nanostack`.
+
+Mbed OS targets should follow the same scheme in CMake.
+
+- `mbed-target-stm32-mcu-stm32l4` - base target
+- `mbed-target-stm32-disco-l4r9i` - final target - a user board or could be custom target, inherits from MCU_  target
+
+If we are not consistent with naming, we would have targets different named than the rest of the tree in Mbed OS and potentionally also in the user space.
```

Comment:
```suggestion
Consistency in naming is a must to avoid having CMake targets with names different than the rest of the Mbed OS repository tree and in the user space.
```